tasks/main.yml

```diff
@@ -627,6 +627,193 @@
         __mssql_conf_setting: "network forceencryption"
         __mssql_conf_setting_value: "{{ '1' if mssql_tls_enable else 'unset' }}"
 
+- name: Configure Active Directory authentication
+  vars:
+    __mssql_kinit_user: >-
+      {{ ad_integration_user }}@{{ ad_integration_realm
+      | upper }}
+  when: mssql_ad_configure | bool
+  block:
+    - name: Join to {{ ad_integration_realm }} realm
+      vars:
+        ad_integration_realm: "{{ ad_integration_realm }}"
+        ad_integration_password: "{{ ad_integration_password }}"
+      include_role:
+        name: linux-system-roles.ad_integration
+
+    - name: Ensure the adutil package
+      package:
+        name: adutil
+        state: present
+
+    - name: List valid kerberos tickets
+      shell: klist || true
+      register: __mssql_klist
+      ignore_errors: true
+      changed_when: false
+
+    - name: kinit as the {{ ad_integration_user }} priviledged user
+      shell: >-
+        echo {{ ad_integration_password | quote }}
+        | kinit {{ __mssql_kinit_user }}
+      when: >-
+        __mssql_kinit_user not in __mssql_klist.stdout
```

Comment:
Even if the principal is in the list - if it is expired, you will have to kinit again

What you really want to do is this:
```
if principal is not present, or principal ticket is expired
  kinit principal
  report changed
else  # principal is present and ticket is current
  do nothing, report no change
fi
```

Reply:
Done that, please review

Reply:
Running `klist -l` prints `(Expired)` for expired tickets, I used that.
```
[root@primary ~]# klist
Ticket cache: KCM:0
Default principal: Administrator@DOMAIN.COM

Valid starting       Expires              Service principal
01/27/2023 17:58:32  01/28/2023 03:54:35  host/primary.domain.com@DOMAIN.COM
	renew until 02/03/2023 17:54:35
01/27/2023 17:54:38  01/28/2023 03:54:35  host/primary.example.com@
	renew until 02/03/2023 17:54:35
	Ticket server: host/primary.example.com@DOMAIN.COM
01/27/2023 17:54:35  01/28/2023 03:54:35  krbtgt/DOMAIN.COM@DOMAIN.COM
	renew until 02/03/2023 17:54:35
01/27/2023 17:54:37  01/28/2023 03:54:35  ldap/ad1.domain.com@DOMAIN.COM
	renew until 02/03/2023 17:54:35
01/27/2023 17:58:40  01/27/2023 18:00:40  kadmin/changepw@DOMAIN.COM
	renew until 01/27/2023 18:00:40
[root@primary ~]# klist -l
Principal name                 Cache name
--------------                 ----------
Administrator@DOMAIN.COM       KCM:0 (Expired)
```